lib/fluent/plugin/out_azure_servicebus_queue.rb

```diff
@@ -58,5 +71,21 @@ def generateToken(url,key_name,key_value)
 
       "SharedAccessSignature sr=#{target_uri}&sig=#{signature}&se=#{expires}&skn=#{key_name}"
     end
+
+    # reference1: https://github.com/microsoft/fluent-plugin-azure-storage-append-blob/blob/master/lib/fluent/plugin/out_azure-storage-append-blob.rb
+    # reference2: https://github.com/Azure/azure-sdk-for-ruby/blob/master/runtime/ms_rest_azure/lib/ms_rest_azure/credentials/msi_token_provider.rb
+    # reference3: https://docs.microsoft.com/en-us/rest/api/servicebus/send-message-to-queue
+    def generateMSIToken(clientid)
+      access_key_request = Faraday.new('http://169.254.169.254/metadata/identity/oauth2/token?' \
```

Comment:
what is the `http://169.254.169.254` endpoint? can we store it as a constant and name for what it is?

Reply:
This is the managed identities for Azure resources endpoint for the Instance Metadata Service. This is the reference link: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token
good point. I will store it as a constant.